### DISCUSSION TITLE
macosx compilation

### DISCUSSION THREAD

**Discussion by neshume** [tgfrerer/island#4] [Ideas]
I have  been trying to compile on a macosx and came across several issues. Have you tried that ? that should not really be an issue... apart from small internal implementations. But I would not like to do extra work if that work has already been done :o)
 and If not I will be willing to put an effort into it.
cheers,

neshume

**Comment by tgfrerer** [comment]
I've not tried to port Island to MacOS - mainly, because Apple doesn't appear keen to support Vulkan. I would want to prioritize Windows (as the platform with more market share) next. Keen to hear how it goes if you want to give MacOS a go, but beware, there be dragons ;)

* On MacOS, Vulkan is emulated via MoltenVK which is part of the [official mac os vulkan sdk](https://vulkan.lunarg.com/sdk/home#mac)
* ~~Island was tested using gcc, and mac uses clang for its compiler toolchain - this might mean some minor tweaks, mainly removing the `--no-gnu-unique` compiler option when compiling using clang~~ edit: that's fixed now. clang is now supported on linux 
* mac os is based on bsd, and bsd is not linux, so it's quite likely that linking and reloading is slightly different, and that hot-reloading might need some tweaks because of that

**Comment by neshume** [comment]
you have some good points but... :o) I believe running this (very good work by the way!!) on mobile (ios + android + other) can really make a difference .. windows is nice but trivial and not that important.. dynamic reload does not need to work on ios/android but on the development platform aka macosx/linux.. I think I will focus, with your blessing of course :o) on macosx/ios/android

**Comment by tgfrerer** [comment]
Oh, having this work on android would be great ;) Android is relatively close to Linux, and Vulkan is pretty well supported there, as far as I know. 

I haven't had a chance to test Island on android because I don't have hardware to test with, but it could be really nice to get it to run on a mobile platform.
